wandb/sdk/lib/sock_client.py

```diff
@@ -51,7 +52,14 @@ def _send_message(self, msg: Any) -> None:
         data = msg.SerializeToString()
         assert len(data) == raw_size, "invalid serialization"
         header = struct.pack("<BI", ord("W"), raw_size)
-        self._sock.sendall(header + data)
+        total_sent = 0
+        message = header + data
+        while total_sent < len(message):
+            with self._lock:
+                sent = self._sock.send(message[total_sent:])
```

Comment:
Might be worth adding a comment on what the default number of bytes this call sends by default.  I was confused when I first read the logic.  I'm also curious how python decides how much to send...

Reply:
it is dependent on the OS...   might be more performant to break it up in the case of very large messages that will likely need to be resent... but it should be based on socket buffer settings.